### DISCUSSION TITLE
Angular 17 @for directive not working with Stepper

### DISCUSSION THREAD

**Discussion by dalvarez-opt** [primefaces/community#1993] [PrimeNG] [answered]
I have recently started using Primeng, latest version, and trying to use new component https://primeng.org/stepper.

If I write fixed html, it renders ok, but when I try to render panels dynamically, it doesn't work.

I've tryed using old *ngFor in p-stepperPanel label, I have tryed using ng-container *ngFor and inside this a p-stepperPanel, I've used new angular @for {}, and nothing works. StepperPanels are not being rendered and I only see empty bindings:

Have a loot at code:

![image](https://github.com/primefaces/community/assets/91538406/82e7a839-f9e9-4e2a-b1b8-73cd66404fc5)

![image](https://github.com/primefaces/community/assets/91538406/3f5aa500-b4b5-4325-863d-35a30edce57c)

Any help would be appreciated

**Comment by dalvarez-opt** [comment]
Hi all. 

I have added a ElementRef to my component.ts and I can see stepperPanels are being added to stepper:

![image](https://github.com/primefaces/community/assets/91538406/b64d7628-95ff-4bfb-b4ec-0df6452f5d0d)

But nothing is being rendered to html.

Any ideas?

**Comment by dalvarez-opt** [comment]
Adding @if at the start of the HTML solved the issue.

**Comment by subcero123** [comment]
Could you provide the corrected code? I'm facing the same issue

**Comment by dalvarez-opt** [comment, marked as answer]
Yes, simply add a @if annotation, so the for loop only fires when steps have been loaded in component.ts:

![image](https://github.com/user-attachments/assets/0703601c-6cc6-4c72-9e09-4872f25ddca8)
